### PR TITLE
NEXUS-4521: SimpleRemoteLeakUT

### PR DESCRIPTION
**Pull request by cstamas**
Test clean up.
- 2nd method was disabled since it was taking too long. Today, it takes 8sec on my mac
- tests cleaned up
- Switched to Hamcrest Matchers.
- removed some junk (unused PlexusTestCase XML config)

### CODE REVIEW

File: nexus/nexus-proxy/src/test/java/org/sonatype/nexus/proxy/SimpleRemoteLeakLRTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import org.hamcrest.MatcherAssert;
+import org.junit.Test;
+import org.sonatype.jettytestsuite.ServletServer;
+import org.sonatype.nexus.proxy.item.RepositoryItemUid;
+import org.sonatype.nexus.proxy.repository.ProxyRepository;
+import org.sonatype.nexus.proxy.repository.RemoteStatus;
+import org.sonatype.nexus.proxy.storage.remote.RemoteStorageContext;
+import org.sonatype.nexus.proxy.storage.remote.commonshttpclient.CommonsHttpClientRemoteStorage;
+
+public class SimpleRemoteLeakLRTest
```
Comment (velo) [inline nexus/nexus-proxy/src/test/java/org/sonatype/nexus/proxy/SimpleRemoteLeakLRTest.java]: Missing jdoc

File: nexus/nexus-proxy/src/test/java/org/sonatype/nexus/proxy/SimpleRemoteLeakLRTest.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    protected EnvironmentBuilder getEnvironmentBuilder()
+        throws Exception
+    {
+        ServletServer ss = (ServletServer) lookup( ServletServer.ROLE );
+        this.jettyTestsuiteEnvironmentBuilder = new M2TestsuiteEnvironmentBuilder( ss );
+        return jettyTestsuiteEnvironmentBuilder;
+    }
+
+    @Test
+    public void simplerAvailabilityCheckRemoteLeak()
```
Comment (peterlynch) [inline nexus/nexus-proxy/src/test/java/org/sonatype/nexus/proxy/SimpleRemoteLeakLRTest.java]: by reading the test name and javadoc ( ie none) I should be able to grok what this is testing - is it just me or this is not possible to do with current name and doc in this LRTest class?

**Comment by peterlynch** [comment]
I doubt test will add 7 more seconds on grid so I'm in favor on second thought of not creating LRTest and leave method where it was, but enabled. Some doc and better test method name would be helpful

**Comment by peterlynch** [comment]
+1

File: nexus/nexus-proxy/src/test/java/org/sonatype/nexus/proxy/SimpleRemoteLeakTest.java [hunk 3]
```diff
@@ -29,6 +33,14 @@ import org.sonatype.nexus.proxy.repository.RemoteStatus;
...
 import org.sonatype.nexus.proxy.storage.remote.commonshttpclient.CommonsHttpClientRemoteStorage;
 
+/**
+ * Httpclient caches connection in a connection pool. This test ensures we are using HttpClient API in correct way,
+ * and that the pool does not introduce a "leak" (ie. after a method is executed, the connection is closed and returned
+ * to pool). This test enforces real HTTP transport to happen multiple times, and checks for pool elements, there should
+ * be no more than 1 connection. Naturally, this test deeply depends on HttpClient 3.x API, so it will work only if
+ * RemoteRepositoryStorage is CommonsHttpClientRemoteStorage. The test is repeated twice, once with "good" transport
+ * and once with "bad" transport (by setting hostname intentionally to a unknown one).
+ */
```
Comment (bdemers) [inline nexus/nexus-proxy/src/test/java/org/sonatype/nexus/proxy/SimpleRemoteLeakTest.java:43]: Nice comment!

**Comment by velo** [comment]
+1

**Comment by bdemers** [comment]
+1

**Pull request merged**